cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/edges/flows/Dataflow.kt

```diff
@@ -63,6 +63,15 @@ class PartialDataflowGranularity(
     val partialTarget: Declaration?
 ) : Granularity
 
+/**
+ * This dataflow granularity denotes that not the "whole" object is flowing from [Dataflow.start] to
+ * [Dataflow.end] but only parts of it. Common examples include tuples or array indices.
+ */
+class IndexedDataflowGranularity(
```

Comment:
Will we then add a third class for partial struct access (e.g. field, offset)?

Reply:
The field-access is already modeled by the `PartialDataflowGranularity` which would receive the `FieldDeclaration` as an input. The offset could probably be used here too (putting the offset as an int as an input). For non-constant offsets, we have nothing matching yet but we can think about this in the future

Reply:
They are probably all of the same type to some degree (tuple deconstruct, array index, struct index) so we should combine them at some point and also model the missing data flows. For example we don't model them for array subscription expressions at the moment.